Automattic/jetpack
# Decouple contact form into a jetpack/forms package

## Proposed changes:


This PR adds a new jetpack/forms package and moves the existing contact form code to it.

As of right now, I've moved most if not all of the PHP/backend code, with form Gutenberg blocks the only thing remaining but we'll need to work on the setup for compiling them first.

I went ahead and introduced the following structure, that I think might make sense going forward:

```
packages/forms/src
    ├- blocks          # Assuming we decouple our blocks from the contact form implementation, that'd go here
    ├- block-editor    # Gutenberg blocks
    ├- contact-form    # Existing contact form implementation
    └- dashboard       # New feedback dashboard implementation
```

I took the opportunity to split up the contact form into code separate classes/files to make it a little easier to find stuff.

**To do's:**

- [ ] Exctract contact form gutenberg blocks from `plugins/jetpack`
- [ ] Add tooling for building a js bundle for the jetpack blocks

### Other information:


## Jetpack product discussion



pcsBup-Fk-p2

## Does this pull request change what data or activity we track or use?



## Testing instructions:






- Run `jetpack install plugins/jetpack`.
- Make sure `JETPACK_BLOCKS_BETA` is set to true, to toggle the package code on.
- There should be no functional changes at all compared to the old implementation.
